src/Data/Parameterized/Vector/NonEmpty.hs

```diff
@@ -0,0 +1,699 @@
+{-# Language GADTs, DataKinds, TypeOperators, BangPatterns #-}
+{-# Language PatternGuards #-}
+{-# Language PolyKinds #-}
+{-# Language TypeApplications, ScopedTypeVariables #-}
+{-# Language TupleSections #-}
+{-# Language Rank2Types, RoleAnnotations #-}
+{-# Language CPP #-}
+#if __GLASGOW_HASKELL__ >= 805
+{-# Language NoStarIsType #-}
+#endif
+{-|
+Copyright        : (c) Galois, Inc 2014-2019
+
+A fixed-size vector of typed elements.
```

Comment:
Mention the non-empty requirement here, and perhaps compare this to `Data.Parameterized.Vector` briefly.

Reply:
Ah, thanks for noticing this